File: 144-BinaryTreePreorderTraversal/144-BinaryTreePreorderTraversal.py
# Last updated: 07/04/2026, 11:41:01
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:

    def tras(self,root,ans):
        if root :
            ans.append(root.val)
            self.tras(root.left,ans)
            self.tras(root.right,ans)
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        self.tras(root,ans)
        return ans
        

File: 144-BinaryTreePreorderTraversal/144-BinaryTreePreorderTraversal.cpp
// Last updated: 07/04/2026, 11:40:57
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> preorderTraversal(TreeNode* root) {
        stack<TreeNode *> st;
        vector<int> ans;
        if ( root == nullptr) return ans;
        st.push(root);
        while(!st.empty()){
        
            TreeNode *node  = st.top();
            st.pop();
            ans.push_back(node->val);
            
            if ( node->right != nullptr) st.push(node->right);
            if ( node->left != nullptr) st.push(node->left);
        
        }
        return ans;

    }
};